ipn/store/kubestore/store_kube.go

```diff
@@ -46,13 +47,21 @@ func (s *Store) ReadState(id ipn.StateKey) ([]byte, error) {
 		}
 		return nil, err
 	}
-	b, ok := secret.Data[string(id)]
+	b, ok := secret.Data[sanitizeKey(id)]
 	if !ok {
 		return nil, ipn.ErrStateNotExist
 	}
 	return b, nil
 }
 
+func sanitizeKey(k ipn.StateKey) string {
+	id := string(k)
+	if strings.Contains(id, "/") {
```

Comment:
Code is easier to read without the if statement, do you need it for performance?

why __? Changing the length is unfortunate and doesn’t actually escape to remove conflicts.

Reply:
picked `__` arbitrarily, went with `_` and more stricter parsing.